internal/zinc-core/src/main/scala/sbt/internal/inc/Incremental.scala

```diff
@@ -340,18 +340,21 @@ object Incremental {
       incremental.invalidateInitial(previous.relations, initialChanges)
 
     // If there's any compilation at all, invalidate all java sources too, so we have access to their type information.
```

Comment:
What about this part?  Does this change lead to a loss of type information?

Reply:
That comment was added in https://github.com/sbt/zinc/commit/0504f70454fdbe9db2cba4003c710888f6e31e2a (port of https://github.com/mspnf/zinc/blob/9ef2eba6f829c2afa783987f64ea4e785b678265/internal/zinc-core/src/main/scala/sbt/internal/inc/Incremental.scala). The type information that this makes it available is to the downstream subproject compilation, which is only relevant in pipelining.

Reply:
Ah, cool, thanks.